src/library/itunes/itunesfeature.cpp

```diff
@@ -402,51 +302,36 @@ TreeItem* ITunesFeature::importLibrary() {
     // configuration we may replace this with something based on the detected
     // location of the user's iTunes path but the defaults are necessary in case
     // their iTunes XML does not include the "Music Folder" key.
-    m_mixxxItunesRoot = "";
-    m_dbItunesRoot = localhost_token();
-
-    //Parse iTunes XML file using SAX (for performance)
-    QFile itunes_file(m_dbfile);
-    if (!itunes_file.open(QIODevice::ReadOnly)) {
-        qDebug() << "Cannot open iTunes music collection";
-        return nullptr;
-    }
+    m_pathMapping.mixxxITunesRoot = "";
+    m_pathMapping.dbITunesRoot = kiTunesLocalhostToken;
 
-    QXmlStreamReader xml(&itunes_file);
-    TreeItem* playlist_root = nullptr;
-    while (!xml.atEnd() && !m_cancelImport) {
-        xml.readNext();
-        if (xml.isStartElement()) {
-            if (xml.name() == QLatin1String("key")) {
-                QString key = xml.readElementText();
-                if (key == "Music Folder") {
-                    if (isTracksParsed) {
-                        isMusicFolderLocatedAfterTracks = true;
-                    }
-                    if (readNextStartElement(xml)) {
-                        guessMusicLibraryMountpoint(xml);
-                    }
-                } else if (key == "Tracks") {
-                    parseTracks(xml);
-                    if (playlist_root != nullptr) {
-                        delete playlist_root;
-                    }
-                    playlist_root = parsePlaylists(xml);
-                    isTracksParsed = true;
-                }
-            }
-        }
+    ITunesImport iTunesImport;
+    std::unique_ptr<ITunesImporter> importer;
+
+#ifdef __MACOS_ITUNES_LIBRARY__
+    if (m_dbfile.isEmpty()) {
+        importer = std::make_unique<ITunesMacOSImporter>(this, m_database, m_cancelImport);
+    } else {
+        importer = std::make_unique<ITunesXMLImporter>(
+                this, m_dbfile, m_database, m_pathMapping, m_cancelImport);
```

Comment:
I think we should log which importer was chosen here. Choosing the importer simply based on whether we found the db file seems somewhat brittle (though idk what else would be more robust).

Reply:
The idea is that e.g. a user on macOS can explicit select an XML file, but also click `Use Default Library` in the context menu to switch back to the native iTunes integration. Perhaps we should make this more explicit?

I agree that logging the importer is probably a good idea.

Reply:
I don't quite understand the workflow, what context menu exactly? Is "native iTunes integration" the XML-file or the iTunesLibrary Framework solution?

Reply:
> Is "native iTunes integration" the XML-file or the iTunesLibrary Framework solution?

The `iTunesLibrary` framework solution.

> I don't quite understand the workflow, what context menu exactly?

![image](https://user-images.githubusercontent.com/30873659/227251463-fb42291a-56c5-4351-b3a4-239f4b8e223a.png)

Selecting `Use Default Library` sets the dbfile path to an empty string which we interpret as "use the `iTunesLibrary` importer". If the dbfile path is set to an XML file, we use the XML importer instead. This should also handle existing users of the XML file integration on macOS gracefully (e.g. someone who has an iTunes XML on an external drive etc.)

Reply:
Mhmm, maybe add a comment that we interpret it that way idk. You're free to improve it in a future PR.

Reply:
I've added a short doc comment to `m_dbfile` and a short log message